rust/private/rust.bzl

```diff
@@ -143,10 +143,7 @@ def _rust_binary_impl(ctx):
     toolchain = find_toolchain(ctx)
     crate_name = ctx.label.name.replace("-", "_")
 
-    if (toolchain.target_arch == "wasm32"):
-        output = ctx.actions.declare_file(ctx.label.name + ".wasm")
-    else:
-        output = ctx.actions.declare_file(ctx.label.name)
+    output = ctx.actions.declare_file(ctx.label.name + toolchain.binary_ext)
```

Comment:
I am surprised WASM test still pass as there is no declared extension corresponding: https://github.com/jelmansouri/rules_rust/blob/win_support/rust/platform/triple_mappings.bzl#L39

I don't really understand the WASM change though so I cannot tell if its missing tests or because the extension is not important.

Reply:
So I have a couple changes coming up after to not use run_shell in the subset of rules we are using. (cc_library and cc_binary). Similarly to this change we did on go:
https://github.com/bazelbuild/rules_go/blob/master/go/private/rules/binary.bzl#L120
It's a great thing to add the CI and I'm really happy to do it!
Our goal is not to rely on genrules nor run_shell on Windows.

Reply:
So what is using the Windows CI to fail is the use of run_shell, my other that I did in our codebase to not use run_shell, needs to be redone given that the command used to run the compiler is gotten more complicated

Reply:
Submitted some WIP to have a green CI, do you think it's ok if I get a similar change as in the go rule linked above in the function I added (will need to shuffle stuff around)?

Reply:
Yes this is fine. I also prefer genrule over run_shell as the later needs a posix shell.